File: du.py
#!/usr/bin/python

import sys
import plib

Args=[]

for arg in sys.argv[1:]:
	import re
	plib.log("pre-ARG:", arg)
	m = re.search('(?<=-)\w+', arg)
	plib.log("arg(", arg, ") m(", m, ")")
	if m:
		if m.group(0).isdigit():
		    arg = "--max-depth=" + m.group(0)
		    plib.log("postARG:", arg)
		if m.group(0) in 'd':
			plib.enable_debug_log()
			arg = ""
	Args.append(arg)

plib.log(sys.argv)
plib.log(Args)

Du = plib.run("/usr/bin/du --bytes", " ".join(Args)).split("\n")
dud = {}
for dul in Du:
	duls = dul.strip()
	if len(duls) > 0:
		size, name = duls.split()
		plib.log("name(", name, ") size(", size, ")")
		dud[name] = int(size)
	
plib.log("dud{", dud, "}")

for name, size in sorted(dud.iteritems(), key=lambda (k,v): (v,k)):
	print plib.add_commas(size).rjust(14), name

File: du.pl
#!/usr/local/bin/perl
foreach my $arg (@ARGV) {
    $arg = "--max-depth=$1" if ($arg =~ /-([0-9])/);
}

@out = `/usr/bin/du --bytes @ARGV`;

@sout = sort lsort @out;
foreach $d (@sout) {
   ($size, $dir) = split(/\s+/,$d);
   printf "%13s : %s\n",addcommas($size),$dir;
}

###############################################################
sub lsort {
   ($a1) = split(/\s+/,$a);
   ($b1) = split(/\s+/,$b);
   #print $a1."<=>".$b1." : ", ($a1) <=> ($b1);
   return (rev_human($a1)) <=> (rev_human($b1));
}

###############################################################
sub addcommas() {
   # put commas in the right places in an integer
   my $input = shift;
   1 while $input =~ s/(\d)(\d\d\d)(?!\d)/$1,$2/g;  
   return $input;
}

###############################################################
sub rev_human() {
	my $num = shift;
	my $s = "";
	if ($num =~ /^([\d\.]+)([GMK])$/) {
		$s = $2;
		$num = $1;
	}
	if ($s =~ /G/) {
		return ($num * 1e9);
	} elsif ($s =~ /M/) {
		return ($num * 1e6);
	} elsif ($s =~ /K/) {
		return ($num * 1e3);
	}
	return $num;
}
